lib/dai-plugin-mcd/src/CollateralService.js

```diff
@@ -0,0 +1,160 @@
+import Maker from '@makerdao/dai';
+import { WAD, RAY } from '@makerdao/dai/utils/constants';
+import BigNumber from 'bignumber.js';
+import { getIlkForCurrency, getSpotContractNameForCurrency, getPipContractNameForCurrency, getFlipContractNameForCurrency } from './utils';
+import CdpType from './CdpType';
+
+export default class CollateralService extends Maker.PublicService {
+
+// Collateral-Specic Data ----------------------------------------------
+
+  constructor(name = 'collateral') {
+    super(name, ['smartContract']);
+  }
+
+  initialize(settings = {}) {
+    if (settings.cdpTypes) {
+      this._cdpTypes = settings.cdpTypes.map( ilk => getCdpTypeObject(ilk.currency));
+    }
+  }
+
+  //todo: allow specifying unit for return value (requires a price service that gives price given an ilkBytes)
+  async getTotalCollateral(currency) {
```

Comment:
it feels to me like there's so much redundancy in these methods, e.g. getIlkForCurrency over and over, that it would be better to move these methods to CdpType, and then just set the commonly-used values in the constructor

Reply:
Yeah good point, I made that update here: https://github.com/makerdao/dai.js/commit/a204b0651264de7690661226e13b995b3984b493